### DISCUSSION TITLE
How to automatically disable Label if its Target is not set or disabled.

### DISCUSSION THREAD

**Discussion by JouriM2** [AvaloniaUI/Avalonia#15959] [Q&A]
Im trying to make `Label` to auto disable in this situations:
* If Target property is not set
* If Target property is set to non-existing name
* If Target is set but target control is disabled

I use next setter in Label ControlTheme:
```XML
	<ControlTheme x:Key="{x:Type Label}" TargetType="Label">
		<Setter Property="IsEnabled">
			<MultiBinding Converter="{x:Static BoolConverters.And}">
				<Binding Path="$self.Target" Converter="{x:Static ObjectConverters.IsNotNull}"/>
				<Binding Path="$self.Target.IsEnabled"/>
			</MultiBinding>
		</Setter>
```

I expect what IsEnabled will be TRUE **only** if target is set (non-null) and trget enabled.
This is working partially: If Target is set to existing control which is disabled - Label is disabled, but if Target is not set at all or is set to non-existing name, Label still enabled.

```XML
<Label x:Name="lbl1" IsTabStop="True" Content="_A label w hotkey linked" Target="lbl3"/>
<Label x:Name="lbl2" Target="lblNotExist">Some _other content unlinked</Label>
<Label x:Name="lbl21">Target is not set content unlinked</Label>
<Label x:Name="lbl3" IsEnabled="False">_Disabled label</Label>
```
looks like:

![изображение](https://github.com/AvaloniaUI/Avalonia/assets/114450272/fde2c48e-6903-40bf-9298-ac8ccb252d66)

What am I doing wrong?

**Comment by stevemonaco** [comment]
The typical way of doing these things without modifying source is to write an attached behavior (or attached property). ie. Listen for `Target` changes, bind `IsEnabled` to `target.IsEnabled`, etc. The access key will always be underlined (without source changes) even when disabled because it's part of `AccessText`'s rendering.

Otherwise, you can do something like:
```xml
<Label IsEnabled="{Binding #edt.IsEnabled}" Content="_A Edit" Target="{Binding #edt}" />
<TextBox IsEnabled="False" x:Name="edt" />
```

**Reply by JouriM2** [reply]
My goal is to do get same result as in your example but without requirements to write (and worst part: to support) huge amount of text in xml.
Anyway, the condition part where IsEnabled checked works!
The part where I try to check if target is not set or ir set to non-existing control dont.

**Comment by jp2masa** [comment]
If you notice, there is a binding error because `$self.Target` is `null` in the second binding path: `$self.Target.IsEnabled`.

Changing it to `<Binding Path="$self.Target.IsEnabled" FallbackValue="False" />` was supposed to work, but there seems to be a bug with `MultiBinding`, maybe due to the binding refactoring, so it doesn't work. I tried to debug it but I'm not very familiar with the new binding system. I noticed that a `true` value was being generated out of nowhere, and fallback value logic is never reached, so it may be a missing feature, I don't know.

Anyway, this alternative is much simpler and should work in your case:

```xaml
<Setter Property="IsEnabled">
    <Binding Path="$self.Target.IsEnabled" FallbackValue="False" />
</Setter>
```

**Reply by JouriM2** [reply]
Where are no errors in compilation (if you mean this by the "binding errors").
Plus: my code works in it second part (as you can see at picture: the first label liked to last one, which is disabled, and successfully disable itself).
The first part, where I try to check target nullability, not working.
